source/extensions/filters/network/postgres_proxy/postgres_decoder.cc

```diff
@@ -198,44 +198,64 @@ bool DecoderImpl::parseHeader(Buffer::Instance& data) {
     ENVOY_LOG(trace, "postgres_proxy: cannot parse message. Need {} bytes in buffer",
               message_len_ + (startup_ ? 0 : 1));
     // Not enough data in the buffer.
-    return false;
+    return Decoder::NeedMoreData;
   }
 
   if (startup_) {
     uint32_t code = data.peekBEInt<uint32_t>(4);
     // Startup message with 1234 in the most significant 16 bits
     // indicate request to encrypt.
     if (code >= 0x04d20000) {
-      ENVOY_LOG(trace, "postgres_proxy: detected encrypted traffic.");
       encrypted_ = true;
-      startup_ = false;
-      incSessionsEncrypted();
+      // Handler for SSLRequest.
+      if (code == 0x04d2162f) {
```

Comment:
Add this ref: https://www.postgresql.org/docs/current/protocol-message-formats.html? SSLRequest "magic number" (code). The value is chosen to contain 1234 in the most significant 16 bits, and 5679 in the least significant 16 bits.

0x04D2 = 1234
0x162F = 5679

Reply:
I will add this. Thanks.

Reply:
Updated.